roles/destroy_vms/tasks/destroy_vms.yml

```diff
@@ -7,18 +7,9 @@
   file:
     path: "{{ item }}"
     state: absent
-  loop: "{{ creation_to_remove.files |  map(attribute='path') | list }}"
-
-- name: List vms
-  community.libvirt.virt:
-    command: list_vms
-  register: vms_list
-
-- name: Get filtered vm list
-  set_fact:
-    vms_to_remove: "{{ [item] + vms_to_remove | default([]) }}"
-  loop: "{{ vms_list.list_vms }}"
-  when: item in hostvars and hostvars[item]['vendor'] | lower == 'kvm'
+  loop: "{{ creation_to_remove.files | map(attribute='path') | select('match', prefix_regex) | list }}"
+  vars:
+    prefix_regex: "^{{ vm_create_scripts }}/?({{ vms_to_remove | join('|') }})_setup_vm.sh$"
```

Comment:
I think we could take this opportunity to rename the `vm_create_scripts` variable, as it can be confusing. (Is this a list of all VM scripts? Is this a function?)
This var configures the _directory_ for VM _create scripts_, so maybe we could add a `_dir` or `_path` suffix to the variable name to make it more descriptive.

Reply:
I agree with @mkochanowski we should be using prefix's and suffix's in var names to keep better track of them.